GoogleForCreators/web-stories-wp
# WAAPI Pause & setCurrentTime Aggregate methods

## Summary
- Converts `playWAAPIAnimations` method to be `WAAPIAnimationMethods.play`.
- Adds `WAAPIAnimationMethods.pause`
- Adds `WAAPIAnimationMethods.setCurrentTime`
- Updates tests and stories to highlight new methods.
- Add polyfill so `pause` & `currentTime` work on safari.

## Relevant Technical Choices
Only way to reset the animation while it's in the middle of playing is to set the current time of the animation back to 0, but just realized  `currentTime` isn't supported by safari 😢. as well as `pause`.
https://developer.mozilla.org/en-US/docs/Web/API/Animation/currentTime
https://developer.mozilla.org/en-US/docs/Web/API/Animation/pause

I think best move is to just polyfill this for now and possibly remove in the future when WAAPI gets better support. Open to suggestions here tho
https://github.com/web-animations/web-animations-js/blob/dev/docs/support.md#features

## To-do
- utilize these methods in the preview grid to abort animations mid-way through.

## User-facing changes
- none

## Testing Instructions
- Added test coverage in unit tests, so make sure unit tests pass for this section
- Go to storybook `dashboard > Story Animation > default` and use the pause and reset buttons.

---

https://app.zenhub.com/workspaces/web-stories-5e94d3aced449034e1e9b226/issues/google/web-stories-wp/1886
